samples/nrf9160/serial_lte_modem/src/main.c

```diff
@@ -29,32 +34,57 @@ void bsd_recoverable_error_handler(uint32_t err)
 	LOG_ERR("bsdlib recoverable error: %u", err);
 }
 
-void start_execute(void)
+static void exit_idle(struct k_work *work)
 {
 	int err;
 
-	LOG_INF("Serial LTE Modem");
-
-	err = modem_info_init();
+	LOG_INF("Exit Idle");
+	gpio_pin_disable_callback(gpio_dev, CONFIG_SLM_INTERFACE_PIN);
+	gpio_remove_callback(gpio_dev, &gpio_cb);
+	/* Do the same as nrf_gpio_cfg_default() */
+	gpio_pin_configure(gpio_dev, CONFIG_SLM_INTERFACE_PIN,
+			GPIO_DIR_IN | GPIO_PUD_NORMAL);
+	/* Restart SLM services */
+	err = slm_at_host_init();
 	if (err) {
-		LOG_ERR("Modem info could not be established: %d", err);
-		return;
+		LOG_ERR("Failed to init at_host: %d", err);
 	}
+}
 
-	modem_info_params_init(&modem_param);
+static void gpio_callback(struct device *dev,
+		     struct gpio_callback *gpio_cb, u32_t pins)
+{
+	k_work_submit(&exit_idle_work);
+}
 
-	err = slm_at_host_init();
-	if (err != 0) {
-		LOG_ERR("Failed to init at_host: %d", err);
+void enter_idle(void)
+{
+	int err;
+
+	gpio_dev = device_get_binding(DT_NORDIC_NRF_GPIO_GPIO_0_LABEL);
```

Comment:
Why not move the GPIO initialization code to `main()` and leave only `gpio_pin_enable_callback()` here, and `gpio_pin_disable_callback()` in `exit_idle()`?

Reply:
Updated as suggested